daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/UpgradesITDev.scala

```diff
@@ -23,29 +24,156 @@ class UpgradesITDev extends AsyncWordSpec with AbstractScriptTest with Inside wi
+        for {
+          name <- c.downField("name").as[String]
+          versions <- c.downField("versions").as[Int]
+          modules <- c
+            .downField("modules")
+            .as(
+              Decoder
+                .decodeList(Decoder.forProduct2("name", "contents") { (x: String, y: String) =>
+                  (x, y)
+                })
+                .map(Map.from _)
+            )
+        } yield {
+          new PackageDefinition(name, versions, modules)
+        }
+    }
+
+  val packagePattern: Regex = "\\{- PACKAGE *\n((?:.|[\r\n])+?)-\\}".r
+
+  def readPackageDefinitions(fileContent: String): Seq[PackageDefinition] = {
+    packagePattern.findAllMatchIn(fileContent).toSeq.map { m =>
+      yaml.parser
+        .parse(m.group(1))
+        .left
+        .map(err => err: Error)
+        .flatMap(_.as[PackageDefinition])
+        .fold(throw _, identity)
+    }
+  }
+
+  def macroDef(n: Int) =
+    s"""#define V$n(code)
+       |#if DU_VERSION == $n
+       |  #define V$n(code) code
+       |#endif
+       |""".stripMargin
+
+  def buildModules(p: PackageDefinition, dir: Path) =
+    p.modules.foreach { case (moduleFullName, content) =>
+      val path = dir.resolve(Paths.get(moduleFullName.replace(".", "/") + ".daml"))
+      val moduleName = moduleFullName.substring(moduleFullName.lastIndexOf(".") + 1)
+      val fileContent = "{-# LANGUAGE CPP #-}\n" + (1 to p.versions)
+        .map(macroDef _)
+        .mkString + "\nmodule " + moduleName + " where\n\n" + content
+      Files.createDirectories(path.getParent());
+      Files.write(path, fileContent.getBytes(StandardCharsets.UTF_8))
+    }
+
+  def buildDar(
+      dir: Path,
+      name: String,
+      version: Int,
+      deps: Seq[String] = Seq.empty,
+      opts: Seq[String] = Seq.empty,
+  ): Path = {
+    writeDamlYaml(dir, name, version, deps)
+    (Seq(
+      damlc.toString,
+      "build",
+      "--project-root",
+      dir.toString,
+    ) ++ opts).! shouldBe 0
+    dir.resolve(s".daml/dist/${name}-$version.0.0.dar")
+  }
+
+  def buildPackages(p: PackageDefinition, tmpDir: Path): Seq[Path] = {
+    val dir = tmpDir.resolve("daml-package-" + p.name)
+    buildModules(p, dir)
+    (1 to p.versions).toSeq.map(version =>
+      buildDar(dir, p.name, version, opts = Seq("--ghc-option", s"-DDU_VERSION=$version"))
+    )
+  }
+
+  def writeDamlYaml(dir: Path, name: String, version: Int, deps: Seq[String] = Seq.empty) = {
+    val fileContent =
+      s"""sdk-version: 0.0.0
+         |build-options: [--target=2.dev]
+         |name: $name
+         |source: .
+         |version: $version.0.0
+         |dependencies:
+         |  - daml-prim
+         |  - daml-stdlib
+         |${deps.map("  - " + _).mkString("\n")}
+         |data-dependencies:
+         |${darFiles.map("  - " + _).mkString("\n")}
+         |module-prefixes:
+         |  upgrades-my-templates-1.0.0: V1
+         |  upgrades-my-templates-2.0.0: V2
```

Comment:
i'm adding these temporarily, the idea is to gradually move the huge tests to the new style

Reply:
They are now gone :)